tests/phpunit/tests/blocks/register.php

```diff
@@ -957,6 +957,37 @@ public function data_register_block_registers_with_args_override_returns_false_w
 		);
 	}
 
+	/**
+	 * Tests registering a block with variations from a PHP file.
+	 *
+	 * @ticket 61280
+	 *
+	 * @covers ::register_block_type_from_metadata
+	 */
+	public function test_register_block_type_from_metadata_with_variations_php_file() {
+		$filter_metadata_registration = static function ( $metadata ) {
+			$metadata['variations'] = 'variations.php';
+			return $metadata;
+		};
+
+		add_filter( 'block_type_metadata', $filter_metadata_registration, 10, 2 );
+		$result = register_block_type_from_metadata(
+			DIR_TESTDATA . '/blocks/notice'
+		);
+		remove_filter( 'block_type_metadata', $filter_metadata_registration );
+
+		$this->assertInstanceOf( 'WP_Block_Type', $result, 'The block was not registered' );
+		$this->assertSame( 'variations.php', $result->variations, 'The block variations are incorrect' );
```

Comment:
That’s rather unwanted behavior as variations is expected to be an array on the block type instance. It should be never set to the file path. I’m also not entirely sure how it works exactly as I would assume it internally calls:

https://github.com/WordPress/wordpress-develop/blob/fa7902c8c85a2d41b801df6717e5067af8396cd3/src/wp-includes/class-wp-block-type.php#L364L367

So technically speaking, calling manually `variations_callback` later in the test wouldn’t be necessary.

Reply:
That's a good point. As we briefly discussed in person, I'll change the implementation to unset `variations` before creating the `WP_Block` instance, and the test to check that the output of `$result->variations` (and/or `->get_variations()` is indeed the array created by the PHP file).

Reply:
https://github.com/WordPress/wordpress-develop/pull/6668/commits/72a0384dd34b99e8a8b55b0b752e7a2949916ac9

Reply:
The new implementation looks and tests great 👍🏻